openllm-core/src/openllm_core/prompt/prompt_template.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+import typing as t
+
+import attr
+
+from .utils import PromptFormatter
+
+default_formatter = PromptFormatter()
+
+@attr.define(slots=True)
+class PromptTemplate:
+  _template: str | None
+  _system_message: str | None
+
+  def __init__(self, template: str | None = None, system_message: str | None = None) -> None:
+    self._template = template
+    self._system_message = system_message
+
+  def _fill_in_variables(self, **kwargs: t.Any) -> str | None:
+    if self._template is None: return self._template
+    template_variables = default_formatter.extract_template_variables(self._template)
+    prompt_variables = {key: '{' + key + '}' if key not in kwargs else (kwargs[key] or '') for key in template_variables}
```

Comment:
This logic doesn't make sense to me. if ke is in kwargs then how can it be None?
```python
variables = {k: '{'+ k + '}' if k not in kwargs else kwargs[k] for k in template_variables}
```

Reply:
it is possible for kwargs[key] to be None (eg. system_message = None).

Reply:
I don't think kwargs['system_message'] should be None.

Reply:
Sounds good. I will take it out